tests/Unit/Domain/Test_BlockLogicalCoordinates.cpp

```diff
@@ -0,0 +1,151 @@
+// Distributed under the MIT License.
+// See LICENSE.txt for details.
+
+#include "tests/Unit/TestingFramework.hpp"
+
+#include <algorithm>
+#include <array>
+#include <cstddef>
+#include <iterator>
+#include <memory>
+#include <vector>
+
+#include "DataStructures/DataVector.hpp"
+#include "DataStructures/Index.hpp"
+#include "DataStructures/Tensor/Tensor.hpp"
+#include "Domain/BlockLogicalCoordinates.hpp"
+#include "Domain/Domain.hpp"
+#include "Domain/DomainHelpers.hpp"
+#include "Domain/ElementId.hpp"
+#include "Domain/ElementLogicalCoordinates.hpp"
+#include "Domain/InitialElementIds.hpp"
+#include "Utilities/Gsl.hpp"
+#include "tests/Unit/TestHelpers.hpp"
+
+// IWYU pragma: no_forward_declare Tensor
+
+namespace {
+
+template <typename TargetFrame>
+void test_block_logical_coordinates() {
+  Domain<3, TargetFrame> domain(
+      maps_for_rectilinear_domains<TargetFrame>(
```

Comment:
This will make both of our lives harder but we'll want to test your function using rotated rectilinear domains as well I think :+1:

Reply:
Ugh.  Should it matter, as long as the maps in the domains are correct?

Reply:
My only worry is about whether `const double up = gsl::at(element_id.segment_ids(), d).endpoint(Side::Upper);` works regardless of the OrientationMap (that is, I'm worried that a Side::Upper might be located at -1 for a flipped block - I would say that if it works correctly in both the aligned case and flipped case then it probably will do fine on arbitrarily rotated ones as well.

Reply:
I think I would like to see some fuzzy tests.  
1) Set up a multi-block domain with all elements at the coarsest level.  Generate a random list of block_id, element_logical_coords, map them to global coords, and compare the computed block_logical_coords to the random generated ones as for the coarsest level  block_logical = element_logical  
2) similar test on a grid that has refined elements, generate a random number of points in each element (including zero); here you test that computed element_logical_coords get you back the random coords

It would be good to test this on the binary domains once they are in...